libraries/NDP/src/NDP.cpp

```diff
@@ -292,6 +281,41 @@ void NDPClass::interrupt_handler() {
   }
 }
 
+int NDPClass::configureClockandCheckSystem()
+{
+   int s;
+
+    printf("MCU, DSP & NN F/w loaded, so lets restart it before proceeding further.\n");
+    s = syntiant_ndp120_tiny_dsp_restart(ndp);
+    if (s) {
+        printf("polling for DSP_RUNNING failed. s=%d\n",s);
+        return 0;
+    } else {
+        printf("DSP is RUNNING and ready.\n");
+    }
+
+    printf("Change the clock \n");
+    configureClock();
+
+    printf("Update the SPI Speed to %d Hz \n", spi_speed_general);
+    SPI1.endTransaction();
+    SPI1.beginTransaction(SPISettings(spi_speed_general, MSBFIRST, SPI_MODE0));
+
+    delay(100); // This delay to ensure the SPI settings is all fine. We can revisit this and reduce/remove it
+    printf("Clock is also reconfigured so lets restart DSP before proceeding further.\n");
```

Comment:
I think logically, this extra dsp restart is superfluous, I suggest we remove it